examples/geojson-vt.js

```diff
@@ -0,0 +1,92 @@
+// NOCOMPILE
+goog.require('ol.Map');
+goog.require('ol.View');
+goog.require('ol.format.GeoJSON');
+goog.require('ol.source.OSM');
+goog.require('ol.source.VectorTile');
+goog.require('ol.layer.Tile');
+goog.require('ol.layer.VectorTile');
+goog.require('ol.tilegrid');
+goog.require('ol.proj');
+
+
+var replacer = function(key, value) {
+  if (value.geometry) {
+    var type;
+    var rawType = value.type;
+    var geometry = value.geometry;
+
+    if (rawType === 1) {
+      type = geometry.length === 1 ? 'Point' : 'MultiPoint';
+    } else if (rawType === 2) {
+      type = geometry.length === 1 ? 'LineString' : 'MultiLineString';
+    } else if (rawType === 3) {
+      type = geometry.length === 1 ? 'Polygon' : 'MultiPolygon';
+    }
+
+    return {
+      'type': 'Feature',
+      'geometry': {
+        'type': type,
+        'coordinates': geometry.length == 1 ? geometry : [geometry]
+      },
+      'properties': value.tags
+    };
+  } else {
+    return value;
+  }
+};
+
+var tilePixels = new ol.proj.Projection({
+  code: 'TILE_PIXELS',
+  units: 'tile-pixels'
+});
+
+var map = new ol.Map({
+  layers: [
+    new ol.layer.Tile({
+      source: new ol.source.OSM()
+    })
+  ],
+  target: 'map',
+  view: new ol.View({
+    center: [0, 0],
+    zoom: 2
+  })
+});
+
+var url = 'data/geojson/countries.geojson';
+fetch(url).then(function(response) {
+  return response.json();
+}).then(function(json) {
+  var tileIndex = geojsonvt(json, {
+    extent: 4096,
+    debug: 1
+  });
+  var vectorSource = new ol.source.VectorTile({
+    format: new ol.format.GeoJSON(),
+    tileGrid: ol.tilegrid.createXYZ(),
+    tilePixelRatio: 16,
+    tileLoadFunction: function(tile, tileCoord) {
+      var format = tile.getFormat();
+      var data = tileIndex.getTile.apply(tileIndex, tileCoord);
+
+      var features = format.readFeatures(
+        JSON.stringify({
+          type: 'FeatureCollection',
+          features: data.features
+        }, replacer));
+      tile.setLoader(function() {
+        tile.setFeatures(features);
+        tile.setProjection(tilePixels);
+      });
+    },
+    tileUrlFunction: function(tileCoord) {
+      return [tileCoord[0], tileCoord[1], -tileCoord[2] - 1];
```

Comment:
At some point, we have to do a hack here anyway. The way it's done here is one. Another would be to use an arbitrary `url` instead of a `tileUrlFunction`, and get use `tile.getTileCoord()` in the `tileLoadFunction`. Yet another would be to use `{z},{x},{y}` as url and use `url.split(',').map(Number)` to get the tile coordinate in the `tileLoadFunction`. What's the least ugly hack in your opinion @tchandelle?